core/src/test/scala/unit/kafka/zk/ZooKeeperTestHarness.scala

```diff
@@ -52,6 +56,22 @@ abstract class ZooKeeperTestHarness extends JUnitSuite with Logging {
     if (zookeeper != null)
       CoreUtils.swallow(zookeeper.shutdown())
     Configuration.setConfiguration(null)
+    assertNoBrokerControllersRunning()
```

Comment:
@ijuma Thank you for the review. Using `System.nanoTime` to track the time taken for `assertNoBrokerControllersRunning`, the total time taken for all the calls for the tests in `core` was around 7.8 seconds. Changing to `@BeforeClass`, `@AfterClass` didn't make much difference even though the number of calls went down from 518 to 178. This could be because the first call is more expensive due to class loading etc. How about we add this check for now, but disable it once we know that the tests are stable? I couldn't find any noticeable difference in the total test time beyond the usual difference between test runs.